plugin/pkg/admission/security/podsecuritypolicy/admission.go

```diff
@@ -0,0 +1,244 @@
+package admission
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/golang/glog"
+
+	admission "k8s.io/kubernetes/pkg/admission"
+	api "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/runtime"
+	sc "k8s.io/kubernetes/pkg/securitycontext"
+	psp "k8s.io/kubernetes/pkg/security/podsecuritypolicy"
+	psputil "k8s.io/kubernetes/pkg/security/podsecuritypolicy/util"
+	"k8s.io/kubernetes/pkg/watch"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+const (
+	PluginName = "PodSecurityPolicy"
+)
+
+func init() {
+	admission.RegisterPlugin(PluginName, func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		plugin := NewPlugin(client)
+		plugin.Run()
+		return plugin, nil
+	})
+}
+
+type constraint struct {
+	*admission.Handler
+	client clientset.Interface
+
+	reflector *cache.Reflector
+	stopChan  chan struct{}
+	store     cache.Store
```

Comment:
No, we don't have more state here.  We have an allocation controller that allocates uids and guids from a pool and annotates the namespace with the values.  At runtime we look for special strategy conditions that indicate it needs pre-allocated values ("RunAsRange" but has no values set in the options).  If we find this we check the namespace for pre-allocated values and populate the strategy.  If we don't have pre-allocated values the strategy is invalid and will fail create.